iceoryx_posh/include/iceoryx_posh/internal/popo/wait_set.inl

```diff
@@ -40,17 +40,93 @@ inline WaitSet<Capacity>::~WaitSet() noexcept
 }
 
 template <uint64_t Capacity>
-template <typename T, typename... Targs>
-inline cxx::expected<WaitSetError> WaitSet<Capacity>::attachEvent(T& eventOrigin, const Targs&... args) noexcept
+template <typename T>
+inline cxx::expected<uint64_t, WaitSetError>
+WaitSet<Capacity>::attachEventImpl(T& eventOrigin,
+                                   const WaitSetHasTriggeredCallback& hasTriggeredCallback,
+                                   const uint64_t eventId,
+                                   const EventInfo::Callback<T>& eventCallback) noexcept
+{
+    if (!hasTriggeredCallback)
+    {
+        return cxx::error<WaitSetError>(WaitSetError::PROVIDED_HAS_TRIGGERED_CALLBACK_IS_UNSET);
+    }
+
+    Trigger possibleLogicallyEqualTrigger(
+        &eventOrigin, hasTriggeredCallback, cxx::MethodCallback<void, uint64_t>(), eventId, Trigger::Callback<T>());
+
+    for (auto& currentTrigger : m_triggerList)
+    {
+        if (currentTrigger.isLogicalEqualTo(possibleLogicallyEqualTrigger))
+        {
+            return cxx::error<WaitSetError>(WaitSetError::EVENT_ALREADY_ATTACHED);
+        }
+    }
+
+    cxx::MethodCallback<void, uint64_t> invalidationCallback = EventAccessor::getInvalidateTriggerMethod(eventOrigin);
+
+    if (!m_triggerList.push_back(
+            Trigger{&eventOrigin, hasTriggeredCallback, invalidationCallback, eventId, eventCallback}))
```

Comment:
I constantly have to rethink the `hasTriggeredCallback` because I always think it's a boolean which tells that a callback has triggered and not a callback to check if the event has triggered.
What do you think about `triggerQueryCallback` (or maybe `queryCallback`)? This line would become
```
Trigger{&eventOrigin, triggerQueryCallback, invalidationCallback, eventId, eventCallback}
```
Doesn't need to be changed for this PR, just as food for thought.